index.js

```diff
@@ -66,26 +66,23 @@ if (process.env.NODE_ENV !== "production" && module.hot) {
 }
 
 module.exports = terria.start({
+    applicationUrl: window.location,
     configUrl: 'config.json',
     shareDataService: new ShareDataService({
         terria: terria
-    })
+    }),
+    beforeRestoreAppState: () => {
```

Comment:
This feature is not yet available in the terriajs version specified in the package.json. Therefore the improvement must have come from somewhere else.

Reply:
It requires this PR from terriajs https://github.com/TerriaJS/terriajs/pull/6342 . I can make a temporary commit to pull in that version for you to test.

Reply:
If adding param `applicationUrl: window.location` to `start` in the main branch only, we will have the same behaviour, which indicates the PR might over kill although there is no side effect introduced.

Reply:
Yes that'll fix the map loading bugs (3dsmooth instead of 3d terrain) but will break share urls that contain data from plugins.

The way to test it is:
- enable this plugin in main - https://github.com/TerriaJS/TerriaMap/blob/8ce8caf8c8a2860ed772c2244861d3dd31a2c663/plugins.ts#L8
- Run the plugin and follow the directions so that it ends up adding an item to workbench called `3d box`
- Create an copy a share link
- Open share link in new tab
- Error popups up saying it cannot load the `3d box` workbench item